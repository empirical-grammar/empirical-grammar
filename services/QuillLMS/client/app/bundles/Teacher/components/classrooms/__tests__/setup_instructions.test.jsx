import React from 'react';
import { shallow } from 'enzyme';

import SetupInstructions from '../setup_instructions'

import { classroomWithStudents, classroomWithoutStudents } from './test_data/test_data'

describe('SetupInstructions component', () => {

  describe('if there are students', () => {
    const wrapper = shallow(
      <SetupInstructions close={() => {}} classroom={classroomWithStudents} />
    );

    it('should render SetupInstructions', () => {
      expect(wrapper).toMatchSnapshot()
    })

    it('should render a link to the student_logins', () => {
      expect(wrapper.find('a').prop('href')).toMatch(`/teachers/classrooms/${classroom.id}/student_logins`)
    })
  })

  describe('if there are no students', () => {
    const wrapper = shallow(
      <SetupInstructions close={() => {}} classroom={classroomWithoutStudents} />
    );

    it('should render SetupInstructions', () => {
      expect(wrapper).toMatchSnapshot()
    })

    it('should render a link to the student_logins', () => {
      expect(wrapper.find('a').prop('href')).toMatch(`${process.env.CDN_URL}/documents/setup_instructions_pdfs/class_code_links.pdf`)
    })
  })
  })

})
