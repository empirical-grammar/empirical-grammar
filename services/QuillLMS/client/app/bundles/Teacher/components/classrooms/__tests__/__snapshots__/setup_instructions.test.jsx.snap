// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SetupInstructions component should render SetupInstructions 1`] = `
<div
  className="create-a-class-modal-content"
>
  <div
    className="create-a-class-modal-body modal-body setup-instructions"
  >
    <h3
      className="title"
    >
      Download student logins and setup instructions
    </h3>
    <p>
      This PDF includes usernames and passwords for each student and instructions for accessing their Quill accounts.
    </p>
    <img
      src="undefined/images/illustrations/setup-instructions-generic.svg"
    />
    <a
      className="quill-button secondary outlined medium"
      download={true}
      href="undefined/documents/setup_instructions_pdfs/class_code_links.pdf"
      target="_blank"
    >
      Download PDF
    </a>
  </div>
  <div
    className="create-a-class-modal-footer with-back-button"
  >
    <button
      className="quill-button secondary outlined medium"
    >
      Back
    </button>
    <button
      className="quill-button primary contained medium"
      onClick={[Function]}
    >
      Done
    </button>
  </div>
</div>
`;
